File: OTHERS/Functions/Easy/The Palindrome Pattern/the-palindrome-pattern.js
process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split("\n").map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {

        let n = parseInt(readLine());

        let arr = [];
        ai = readLine().trim().split(' ').map((x) => parseFloat(x));
        for (let i = 0; i < n * n; i += n) {
            let j = i, k = n;
            let a = [];
            while (k--) {
                a.push(ai[j]);
                j++;
            }
            arr.push(a);
        }
        let obj = new Solution();
        let res = obj.pattern(arr);
        console.log(res);
        console.log("~");
    }
}

class Solution {
    /**
    * @param number n
    * @param number[][] arr

    * @returns string
    */
    pattern(arr) {
        const n = arr.length;
        for (let i = 0; i < n; i++) {
            let isPalindrome = true;
            for (let j = 0; j < n / 2; j++) {
                if (arr[i][j] !== arr[i][n - j - 1]) {
                    isPalindrome = false;
                    break;
                }
            }
            if (isPalindrome) return `${i} R`;
        }
        for (let j = 0; j < n; j++) {
            let isPalindrome = true;
            for (let i = 0; i < n / 2; i++) {
                if (arr[i][j] !== arr[n - i - 1][j]) {
                    isPalindrome = false;
                    break;
                }
            }
            if (isPalindrome) return `${j} C`;
        }
        return '-1';
    }
}

File: OTHERS/Functions/Easy/The Palindrome Pattern/the-palindrome-pattern.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:

    string pattern(vector<vector<int>> &arr) {
        int n = arr.size();
        for (int i = 0; i < n; i++) {
            bool isPalindrome = true;
            for (int j = 0; j < n / 2; j++) {
                if (arr[i][j] != arr[i][n - j -1]) {
                    isPalindrome = false;
                    break;
                }
            }
            if (isPalindrome) return to_string(i) + " R";
        }
        for (int j = 0; j < n; j++) {
            bool isPalindrome = true;
            for (int i = 0; i < n / 2; i++) {
                if (arr[i][j] != arr[n - i - 1][j]) {
                    isPalindrome = false;
                    break;
                }
            }
            if (isPalindrome) return to_string(j) + " C";
        }
        return "-1";
    }
};

int main() {

    int t, n, i, j, flag, k;
    cin >> t;

    while (t--) {
        cin >> n;
        vector<vector<int>> a(n, vector<int>(n));

        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++)
                cin >> a[i][j];
        }
        Solution ob;
        cout << ob.pattern(a) << endl << "~" << endl;
    }
    return 0;
}